src/libs/Xinterface/Stringservice/strservice.cpp

```diff
@@ -1425,24 +1425,24 @@ uint32_t _DialogAssembleStr(VS_STACK *pS)
         return IFUNCRESULT_FAILED;
     const char *pcID = pDat->GetString();
 
-    char param[4096];
+    std::vector<char> param;
```

Comment:
Because std::string contains const char*, that is not fully suitable for buffer (because of const nature)

Reply:
The only problem with a vector is that it doesn't automaticaly add null terminator to the end, but that is not different from using raw char arrays)

Reply:
Oh, I forgot that basic_string have data() method too, that can return non-const array, so the comments I write before is no longer relevant xD
Maybe it will be better to replace vector by string